code/parse/sexp.cpp

```diff
@@ -20289,7 +20291,7 @@ void sexp_string_concatenate(int n)
 	// check length
 	if (strlen(new_text) >= TOKEN_LENGTH)
 	{
-		Warning(LOCATION, "Concatenated string is too long and will be truncated.");
+		Warning(LOCATION, "Concatenated string '%s' is too long and will be truncated.", new_text);
```

Comment:
Along the same lines as @asarium pointed out below, having this warning display the actual length would be helpful.